openapi/tests/openapi_integration/test_collection_exists.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from .helpers.collection_setup import basic_collection_setup, drop_collection
+from .helpers.helpers import request_with_validation
+
+default_name = ""
+collection_name = "test_collection_exist"
+
+
+@pytest.fixture(autouse=True)
+def setup(on_disk_vectors):
+    basic_collection_setup(
+        collection_name=collection_name, on_disk_vectors=on_disk_vectors
+    )
+    yield
+    drop_collection(collection_name=collection_name)
+
+
+def test_collection_exists():
+    response = request_with_validation(
+        api="/collections/{collection_name}/exists",
+        method="GET",
+        path_params={"collection_name": collection_name},
+    )
+    assert response.ok
+    result = response.json()["result"]["exists"]
+    assert result == True
+
+
+def test_collection_not_found():
+    response = request_with_validation(
+        api="/collections/{collection_name}/exists",
+        method="GET",
+        path_params={"collection_name": "not_found"},
+    )
+    assert response.status_code == 404
+    assert (
+        response.json()["status"]["error"] == "Not found: Collection `not_found` doesn't exist!"
+    )
```

Comment:
I wonder... this returns a 404 now. That is of course the expected REST response, but by this logic you could also simply use `GET /collections/something -> 404`.

I wonder if we should return a proper object here as well with HTTP 200 that has `exists: false`.

@coszio What do you think?

Reply:
Yes, I think the whole point of this API is to make it easier to create functions in clients without relying on `404` error response.